ailab/db/finesse/test_queries/__init__.py

```diff
@@ -1,12 +1,19 @@
 def get_random_chunk(cursor):
     query = """
-        SELECT dc.score AS score, cr.id AS crawl_id, ch.id AS chunk_id, ch.title, cr.url AS crawl_url, ch.text_content, ch.text_content
+        SELECT dc.score AS score, cr.id AS crawl_id, ch.id AS chunk_id, ch.title, cr.url AS crawl_url, ch.text_content
         FROM Chunk ch
         INNER JOIN html_content_to_chunk hctc ON ch.id = hctc.chunk_id
         INNER JOIN html_content hc ON hctc.md5hash = hc.md5hash
         INNER JOIN crawl cr ON hc.md5hash = cr.md5hash
         INNER JOIN documents dc ON ch.id = dc.chunk_id
-        WHERE dc.score > 0.0
+        WHERE dc.score > 0.0 
+        AND EXISTS (
+            SELECT 1 
+            FROM score sc
+            WHERE sc.entity_id = ch.id
+            AND sc.score_type = 'current' 
+            AND sc.score > 0.0
+        )
         ORDER BY RANDOM()
```

Comment:
sorting through all the table to pick up a single element is enormously expensive.

how about:

```
SELECT
	*
FROM
	table_name OFFSET floor(random() * (
		SELECT
			COUNT(*)
			FROM table_name))
LIMIT 1;
```

https://www.postgresql.org/docs/current/queries-limit.html

Reply:
@JolanThomassin

Reply:
still not addressed @JolanThomassin

Reply:
I changed it, it look a little faster now, but the quality stay the same.

Reply:
It's definitely way faster.